### PR TITLE
Adding methods and property to manually control write transactions

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Write transactions are currently only done through the `Realm.write()` function. Sometimes it is useful to be able to control the write transactions manually. This PR adds a set of methods.

See also #715, #815, and #1188.

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [x] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -452,6 +453,21 @@ declare class Realm {
      * @returns void
      */
     write(callback: () => void): void;
+
+    /**
+     * @returns void
```
Comment (kristiandupont) [inline lib/index.d.ts]: We should add the comments from the documentation .js file here as well -- hopefully at some point, this file can be the single source of documentation both for the website and for typescript-aware ide's

Comment (kneth) [inline lib/index.d.ts]: Isn't it better do redo/unite the documentation in a separate PR?

Comment (kristiandupont) [inline lib/index.d.ts]: If this applies to other methods than just this one, then yes. I didn't check for that. :-)

**Comment by kneth** [comment]
@kristiandupont @blagoev I have been through the PR feedback, and I believe it is ready to be merged.

**Pull request merged**